pkg/ottl/functions.go

```diff
@@ -74,36 +75,39 @@ func (p *basePath[K]) Next() Path[K] {
 	return p.nextPath
 }
 
-func (p *basePath[K]) Key() Key[K] {
-	if p.key == nil {
+func (p *basePath[K]) Keys() []Key[K] {
+	if p.keys == nil {
 		return nil
 	}
-	return p.key
+	p.fetchedKeys = true
+	return p.keys
 }
 
 func (p *basePath[K]) isComplete() error {
 	if !p.fetched {
 		return fmt.Errorf("the path section %q was not used by the context - this likely means you are using extra path sections", p.name)
 	}
+	if p.keys != nil && !p.fetchedKeys {
+		return fmt.Errorf("the keys were not used by the context - this likely means you are trying to index a path that does not support indexing")
```

Comment:
Could we print something here that points the user to where the error occurred? Either the keys or path would work. I see that we have the path section available here, maybe that's the quickest approach for now.

Reply:
Updated